server/manager/src/main/java/org/apache/accumulo/manager/upgrade/UpgradeCoordinator.java

```diff
@@ -105,8 +106,11 @@ public boolean isParentLevelUpgraded(KeyExtent extent) {
   private static Logger log = LoggerFactory.getLogger(UpgradeCoordinator.class);
 
   private int currentVersion;
-  private Map<Integer,Upgrader> upgraders = Map.of(AccumuloDataVersion.SHORTEN_RFILE_KEYS,
-      new Upgrader8to9(), AccumuloDataVersion.CRYPTO_CHANGES, new Upgrader9to10());
+  // map of "current version" -> upgrader to next version.
+  private Map<Integer,
```

Comment:
`Map.of` returns an unmodifiable map. If you use `final` here, then the `upgraders` variable can't be reassigned. This should give you confidence in the map contents in the code below.